packages/search/src/Search.ts

```diff
@@ -72,6 +75,9 @@ export class Search extends Textfield {
 
     private handleKeydown(event: KeyboardEvent): void {
         const { code } = event;
+        if (code === 'Escape' && this.holdValueOnEscape) {
+            return;
+        }
```

Comment:
Rather than two `if` conditions can we achieve this using a single if condition like this
```
if(code === 'Escape' && !this.holdValueOnEscape)
{
    this.reset();
}
return;
```